### PR TITLE
Update Issue and PR templates

### PR DESCRIPTION
**Pull request by sdzhepa**
### Description (*)
This PR contains 2 improvements/updates

#### 1. Updated Issue related templates
Added section that allows a reporter to set initial Severity during issue reporting as it described on DevDoc  [Who and how can define severity and priority](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#who-and-how-can-define-severity-and-priority) 
```
Issue reporters can provide their own evaluation for severity by selecting a checkbox in the Issue description.
```

#### 2. Improve Pull Request templates
Added keyword to the section Fixed Issue
```
Fixes magento/magento2#<issue_number>
```
This keyword should improve automatically linking PR -> Issue, according to GitHub documentation [Linking a pull request to an issue using a keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### CODE REVIEW

**Comment by sdzhepa** [comment]
@magento run all test

**Comment by sdzhepa** [comment]
@magento run all tests

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -32,3 +32,12 @@ Important: Provide a set of clear steps to reproduce this bug. We can not provid
 <!--- Tell us what happened instead. Include error messages and issues. -->
 1. [Screenshots, logs or description]
 2.
+
+---
+Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.
+
+- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users with no workaround._
```
Comment (orlangur) [inline .github/ISSUE_TEMPLATE/bug_report.md]: "without workaround" maybe?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -23,7 +23,7 @@
     If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
     There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
 -->
-1. magento/magento2#<issue_number>: Issue title
+1. Fixes magento/magento2#<issue_number>
```
Comment (orlangur) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: 👍

**Review by orlangur — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (sdzhepa) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed, thank you

**Review by ihor-sviziev — APPROVED** [review]
Approved ✔
Test failures are definitely not related to changes in this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7537](https://jira.corp.magento.com/browse/ENGCOM-7537) has been created to process this Pull Request

**Review by okorshenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @okorshenko, thank you for the review.
[ENGCOM-7537](https://jira.corp.magento.com/browse/ENGCOM-7537) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Charlie** [comment]
Failed functional tests not related to changes from this PR.

**Pull request merged**